frontend/sac-mobile/app/(auth)/register.tsx

```diff
@@ -37,114 +65,259 @@ const Register = () => {
 
     const onSubmit = (data: RegisterFormData) => {
         try {
-            registerSchema.parse(data);
-            Alert.alert('Form Submitted', JSON.stringify(data));
+            const { graduationYear, passwordConfirm, ...rest } = data;
+            const updatedData = {
+                ...rest,
+                graduationYear: graduationYear.value
+            };
+            registerSchema.parse(updatedData);
+            Alert.alert('Form Submitted', JSON.stringify(updatedData));
             router.push('/(app)/');
         } catch (error) {
             if (error instanceof ZodError) {
                 Alert.alert('Validation Error', error.errors[0].message);
-            } else {
+            }
+            else {
                 console.error('An unexpected error occurred:', error);
             }
         }
     };
 
     return (
-        <View className="items-center justify-center flex-1 p-4">
-            <View className="w-full mb-4">
-                <Text>First Name</Text>
-                <Controller
-                    control={control}
-                    render={({ field: { onChange, value } }) => (
-                        <TextInput
-                            className="p-2 border border-gray-300"
-                            placeholder="Ladley"
-                            onChangeText={onChange}
-                            value={value}
-                            onSubmitEditing={handleSubmit(onSubmit)}
-                        />
-                    )}
-                    name="firstName"
-                    rules={{ required: 'First name is required' }}
+        <SafeAreaView
+            className="bg-neutral-500"
+            edges={['top']}
+        >
+            <ScrollView>
+            <View className="px-[8%] pb-[9%]">
+                <Wordmark
+                    textColor="white"
+                    button={true}
+                    func={() => router.push('/(auth)/login')}
+                    title="Login"
                 />
-                {errors.firstName && <Text>{errors.firstName.message}</Text>}
+                <View className="pt-[9%] pb-[6%]">
+                    <Header text="Sign up" fontSize="45" color="white"></Header>
```

Comment:
- sizes should be static, `sm`, `md`, `lg`, etc 
- should be either text as children rather than as a text prop

Reply:
but also why a header component over just using the native wind styling since its only 3 styles may be unncessary